### PR TITLE
nmd 490 add controllers

### PR DESCRIPTION
**Pull request by VadimZharikov**
mapper config

add controllers

resolve

sync

add models, controllers, configure mapper

resolve

add generic service

make generic repo delete and getbyid return type to nullable entity

### CODE REVIEW

File: Client/EnlightenmentApp.API/Enums/TagTypeEnum.cs [hunk 1]
```diff
@@ -0,0 +1,12 @@
+﻿namespace EnlightenmentApp.API.Enums
+{
+    public enum TagType
```
Comment (Glipchik) [inline Client/EnlightenmentApp.API/Enums/TagTypeEnum.cs:3]: Let´s move all the enums to separate Domain layer and use it from all other layers

File: Client/EnlightenmentApp.API/Controllers/ChapterController.cs [hunk 2]
```diff
@@ -0,0 +1,63 @@
+﻿using AutoMapper;
+using EnlightenmentApp.API.Models.Chapter;
+using EnlightenmentApp.BLL.Entities;
+using EnlightenmentApp.BLL.Interfaces.Services;
+using Microsoft.AspNetCore.Mvc;
+
+namespace EnlightenmentApp.API.Controllers
+{
+    [Route("api/Chapters")]
+    [ApiController]
```
Comment (Genresal) [inline Client/EnlightenmentApp.API/Controllers/ChapterController.cs:10]: possible **[Route("api/chapters")]** is better

File: Client/EnlightenmentApp.API/Controllers/ChapterController.cs [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+        private IChapterService _chapterService;
+        private IMapper _mapper;
+
+        public ChapterController(IChapterService chapterService, IMapper mapper)
+        {
+            this._chapterService = chapterService;
+            this._mapper = mapper;
+        }
+
+        // GET: api/<Controller>/5
```
Comment (Glipchik) [inline Client/EnlightenmentApp.API/Controllers/ChapterController.cs:22]: Replace simple comments with the controller description notation

**Review by Glipchik — CHANGES_REQUESTED** [review]

**Review by Glipchik — APPROVED** [review]

File: Client/EnlightenmentApp.API/Program.cs [hunk 4]
```diff
@@ -22,6 +31,7 @@
 // Configure the HTTP request pipeline.
 if (app.Environment.IsDevelopment())
 {
+    //app.UseSwagger();
```
Comment (Finka95) [inline Client/EnlightenmentApp.API/Program.cs:34]: Remove this

**Review by Finka95 — APPROVED** [review]

**Pull request merged**